### DISCUSSION TITLE
Is Support Hash Map ?

### DISCUSSION THREAD

**Discussion by kecci** [jellydator/ttlcache#160] [Q&A]
I wonder, is ttlcache support hash map ? such as HSET & HGETALL in redis.

Thanks.

**Comment by swithek** [comment]
Hi, can you give an example of the feature you're looking for?

**Reply by kecci** [reply]
Hi @swithek , thanks for reply.

In my case, I want to build a collection of `map[string]*ttlcache.Cache[string, string]` so I can manage a key with multiple fields cache with ttl per-field.

For example in my code, currently i wrap up like this :
```
type inMemoryCache struct {
  myCacheMutex *sync.RWMutex
  myCache            map[string]*ttlcache.Cache[string, string]
}
```

(in this case, i need to setup the mutex also for my map for read and write)

which this can be more helpful if this could be managed by ttlcache, rather than I create my own wrapper (if possible).

edited: Also I need to create func to Get and Set the map[string] to initiate the ttlcache.New also (one-by-one).

I can show you easily to imagine my example structure map could look like this:
```
- "AAPL"
   - "price" : 1000 (TTL 5s)
   - "volume" : 20 (TTL 1m)
- "GOOG"
   - "price" : 2000 (TTL 5s)
   - "volume" : 30 (TTL 1m)
```
